src/cmd/sanity_test/sanity_tests.go

```diff
@@ -323,91 +328,120 @@ func checkOnedriveRestoration(
 			fileSizes[itemName] = ptr.Val(driveItem.GetSize())
 		}
 
-		folderPermission[itemID] = permissionsIn(ctx, client, driveID, itemID, folderPermission[itemID])
+		if driveItem.GetFolder() == nil && driveItem.GetPackage() == nil {
+			continue
+		}
+
+		// currently we don't restore blank folders.
+		// skip permission check for empty folders
+		if ptr.Val(driveItem.GetFolder().GetChildCount()) == 0 {
+			fmt.Println("skipped empty folder: ", itemName)
+			continue
+		}
+
+		permissionIn(ctx, client, driveID, itemID, itemName, folderPermission)
+		getOneDriveChildFolder(ctx, client, driveID, itemID, itemName, fileSizes, folderPermission, startTime)
+	}
+
+	getRestoreData(ctx, client, *drive.GetId(), restoreFolderID, restoreFile, restoreFolderPermission, startTime)
+
+	for checkFolderName, checkfolderPer := range folderPermission {
+		fmt.Printf("checking for folder: %s \n", checkFolderName)
+
+		if len(checkfolderPer) < 1 {
+			fmt.Printf("no permissions found for folder : %s.", checkFolderName)
+			continue
+		}
+
+		if len(restoreFolderPermission[checkFolderName]) < 1 {
+			fmt.Printf("permissions are not equal")
+			fmt.Println("Item:", checkFolderName)
+			fmt.Println("Permission found: ", checkfolderPer)
+			fmt.Println("blank permission found in restore.")
+
+			os.Exit(1)
+		}
+
+		for i, orginalFolderPer := range checkfolderPer {
+			if !(orginalFolderPer.entityID != restoreFolderPermission[checkFolderName][i].entityID) &&
```

Comment:
Nit: repeated indexes on a value should be declared earlier for readability.
```suggestion
			restoredPemission := restoreFolderPermission[checkFolderName][i]
			
			if !(originalPermission.entityID != restoredPemission.entityID) &&
```